### PR TITLE
[Doc] Add headless section in pages components

### PR DESCRIPTION
**Pull request by fzaninotto**
- [x] Add headless section to Create
- [x] Add headless section to Show
- [x] Add headless section to Edit
- [x] Add headless section to List
- [x] Add headless section to SimpleForm
- [x] Add headless section to Features
- ~~[ ] Add links to demos~~ (no public demo yet)

image

### CODE REVIEW

**Review by djhi — COMMENTED** [review]
Might be worth mentionning this can be used when you want a custom UI without MUI

File: docs/Features.md [hunk 1]
```diff
@@ -271,6 +271,93 @@ We have made many improvements to this default layout based on user feedback. In
 
 And for mobile users, react-admin renders a different layout with larger margins and less information density (see [Responsive](#responsive)).
 
+## Headless
+
+React-admin components use Material UI components by default, which lets you scaffold a page in no time. However, the headless logic behind react-admin components is agnostic of the UI library, and is exposed via -Base components and controller hooks.
```
Comment (adguernier) [inline docs/Features.md:276]: The first time I read, I dit not undertand the meaning of "-Base components" 
```suggestion
React-admin components use Material UI components by default, which lets you scaffold a page in no time. However, the headless logic behind react-admin components is agnostic of the UI library, and is exposed via `...Base` components and controller hooks. 
```

File: docs/InfiniteList.md [hunk 2]
```diff
@@ -161,4 +161,111 @@ export const BookList = () => (
...
+
+## Controlled Mode
+
+`<InfiniteList>` deduces the resource and the list parameters from the URL. This is fine for a page showing a single list of records, but if you need to display more than one list in a page, you probably want to define the list parameters yourself. 
+
+In that case, use the [`resource`](#resource), [`sort`](#sort), and [`filter`](#filter-permanent-filter) props to set the list parameters.
+
+{% raw %}
+```jsx
+import { InfiniteList, SimpleList } from 'react-admin';
```
Comment (adguernier) [inline docs/InfiniteList.md:174]:
```suggestion
import { InfiniteList, InfinitePagination, SimpleList } from 'react-admin';
```

File: docs/InfiniteList.md [hunk 3]
```diff
@@ -161,4 +161,111 @@ export const BookList = () => (
...
+)
+```
+{% endraw %}
+
+## Headless Version
+
+Besides fetching a list of records from the data provider, `<InfiniteList>` renders the default list page layout (title, buttons, filters, a Material-UI `<Card>`, infinite pagination) and its children. If you need a custom list layout, you may prefer the `<InfiniteListBase>` component, which only renders its children in a [`ListContext`](./useListContext.md).
+
+```jsx
+import { InfiniteListBase, InfinitePagination } from 'react-admin';
```
Comment (adguernier) [inline docs/InfiniteList.md:216]:
```suggestion
import { InfiniteListBase, InfinitePagination, WithListContext } from 'react-admin';
```

File: docs/List.md [hunk 4]
```diff
@@ -1081,3 +1081,158 @@ const ProductList = () => (
...
+{% endraw %}
+
+**Note**: If you render more than one `<Datagrid>` for the same resource in the same page, they will share the selection state (i.e. the checked checkboxes). This is a design choice because if row selection is not tied to a resource, then when a user deletes a record it may remain selected without any ability to unselect it. You can get rid of the checkboxes by setting `<Datagrid bulkActionButtons={false}>`.
+
+## Headless Version
+
+Besides fetching a list of records from the data provider, `<List>` renders the default list page layout (title, buttons, filters, a Material-UI `<Card>`, pagination) and its children. If you need a custom list layout, you may prefer [the `<ListBase>` component](./ListBase.md), which only renders its children in a [`ListContext`](./useListContext.md).
+
+```jsx
+import { ListBase } from 'react-admin';
```
Comment (adguernier) [inline docs/List.md:1180]:
```suggestion
import { ListBase, WithListContext } from 'react-admin';
```

File: docs/Show.md [hunk 5]
```diff
@@ -604,31 +604,87 @@ export const PostShow = () => (
...
-The root component of `<Show>` is a Material UI `<Card>`. Besides, `<Show>` renders an action toolbar, and sets the page title. This may be useless if you have a completely custom layout.
+`<show>` deduces the resource and the record id from the URL. This is fine for a detail page, but if you need to embed the details of a record in another page, you probably want to define these parameters yourself. 
 
-In that case, opt for [the `<ShowBase>` component](./ShowBase.md), a headless version of `<Show>`.
+In that case, use the [`resource`](#resource) and [`id`](#id) props to set the show parameters regardless of the URL.
 
 ```jsx
-// in src/posts.jsx
-import { ShowBase } from 'react-admin';
+import { Show, SelectField, SimpleShowLayout, TextField, Title } from "react-admin";
```
Comment (adguernier) [inline docs/Show.md:614]:
```suggestion
import { Show, SelectField, SimpleShowLayout, TextField } from "react-admin";
```

File: docs/useCreateController.md [hunk 6]
```diff
@@ -5,44 +5,47 @@ title: "The useCreateController hook"
 
 # `useCreateController`
 
-The `useCreateController` hook contains the logic of [the `<Create>` component](./Create.md): it prepares a form submit handler, and returns the data and callbacks necessary to render a Creation view. 
+`useCreateController` contains the headless logic of the [`<Create>`](./Create.md) component. It's useful to create a custom creation view. It's also the base hook when building a custom view with another UI kit than Material UI. 
 
-React-admin calls `useCreateController` internally, when you use the `<Create>`, or `<CreateBase>` component.
+`useCreateController` reads the resource name from the resource context and browser location, computes the form default values, prepares a form submit handler based on `dataProvider.create()`, computes the default page title, and returns them. Its return value matches the [`CreateContext`](./useCreateContext.md) shape. 
+
+`useCreateController` is used internally by [`<Create>`](./Create.md) and [`<CreateBase>`](./CreateBase.md). If your Create view uses react-admin components like `<SimpleForm>`, prefer [`<CreateBase>`](./CreateBase.md) to `useCreateController` as it takes care of creating a `<CreateContext>`.
```
Comment (adguernier) [inline docs/useCreateController.md:12]:
```suggestion
`useCreateController` is used internally by [`<Create>`](./Create.md) and [`<CreateBase>`](./CreateBase.md). If your Create view uses react-admin components like [`<SimpleForm>`](./SimpleForm.md), prefer [`<CreateBase>`](./CreateBase.md) to `useCreateController` as it takes care of creating a `<CreateContext>`.
```

File: docs/useEditController.md [hunk 7]
```diff
@@ -5,17 +5,17 @@ title: "The useEditController hook"
...
+`useEditController` contains the headless logic of the [`<Edit>`](./Edit.md) component. It's useful to create a custom edition view. It's also the base hook when building a custom view with another UI kit than Material UI. 
 
-React-admin calls `useEditController` internally when you use the `<Edit>`, `<EditBase>`, or `<EditGuesser>` component.
+`useEditController` reads the resource name and id from the resource context and browser location, fetches the record via `dataProvider.getOne()` to initialize the form, prepares a form submit handler based on `dataProvider.update()`, computes the default page title, and returns them. Its return value matches the [`EditContext`](./useEditContext.md) shape. 
+
+`useEditController` is used internally by [`<Edit>`](./Edit.md) and [`<EditBase>`](./EditBase.md). If your Edit view uses react-admin components like `<SimpleForm>`, prefer [`<EditBase>`](./EditBase.md) to `useEditController` as it takes care of creating a `<EditContext>`.
```
Comment (adguernier) [inline docs/useEditController.md:12]:
```suggestion
`useEditController` is used internally by [`<Edit>`](./Edit.md) and [`<EditBase>`](./EditBase.md). If your Edit view uses react-admin components like [`<SimpleForm>`](./SimpleForm.md), prefer [`<EditBase>`](./EditBase.md) to `useEditController` as it takes care of creating a `<EditContext>`.
```

File: docs/useListController.md [hunk 8]
```diff
@@ -5,13 +5,89 @@ title: "useListController"
...
+`useListController` contains the headless logic of the [`<List>`](./List.md) component. It's useful to create a custom List view. It's also the base hook when building a custom view with another UI kit than Material UI. 
 
-You can use it to create a custom List view, although its component counterpart, [`<ListBase>`](./ListBase.md), is probably better in most cases.
+![List view built with Ant Design](./img/list_ant_design.png)
+
+`useListController` reads the list parameters from the URL, calls `dataProvider.getList()`, prepares callbacks for modifying the pagination, filters, sort and selection, and returns them together with the data. Its return value matches the [`ListContext`](./useListContext.md) shape. 
+
+`useListController` is used internally by [`<List>`](./List.md) and [`<ListBase>`](./ListBase.md). If your list view uses react-admin components like `<Datagrid>`, prefer [`<ListBase>`](./ListBase.md) to `useListController` as it takes care of creating a `<ListContext>`.
```
Comment (adguernier) [inline docs/useListController.md:14]:
```suggestion
`useListController` is used internally by [`<List>`](./List.md) and [`<ListBase>`](./ListBase.md). If your list view uses react-admin components like [`<Datagrid>`](./Datagrid.md), prefer [`<ListBase>`](./ListBase.md) to `useListController` as it takes care of creating a `<ListContext>`.
```

**Comment by fzaninotto** [comment]
Review applied

**Review by adguernier — APPROVED** [review]

File: docs/Create.md [hunk 9]
```diff
@@ -599,3 +601,90 @@ export default OrderEdit;
 ```
 
 **Tip:** If you'd like to avoid creating an intermediate component like `<CityInput>`, or are using an `<ArrayInput>`, you can use the [`<FormDataConsumer>`](./Inputs.md#linking-two-inputs) component as an alternative.
+
+## Controlled Mode
```
Comment (djhi) [inline docs/Create.md:605]: Why *Controlled Mode*? This is misleading imo. Controlled vs Uncontrolled refers to other conceps

Comment (fzaninotto) [reply docs/Create.md:605]: I have no better name for it...

Comment (djhi) [reply docs/Create.md:605]: With Default UI ?

Comment (fzaninotto) [reply docs/Create.md:605]: I don't understand what that means... I don't think it's better than "controlled mode"

**Review by slax57 — APPROVED** [review]

File: docs/Create.md (hunk 9 above)
Comment (djhi) [reply docs/Create.md:605]: nvm

**Review by djhi — APPROVED** [review]

**Pull request merged**